wkcuber/wkcuber/upload.py

```diff
@@ -1,72 +1,64 @@
 
-from ._internal.utils import (
-    add_verbose_flag,
-    parse_path,
-    setup_logging,
-    setup_warnings,
-    AsciiArgumentParser,
-)
-
-
-def create_parser() -> ArgumentParser:
-    parser = AsciiArgumentParser()
-
-    parser.add_argument(
-        "source_path",
-        help="Directory containing the source WKW dataset.",
-        type=parse_path,
-    )
-
-    parser.add_argument(
-        "--token",
-        help="Auth token of the user on WEBKNOSSOS",
-        default=None,
-    )
+app = typer.Typer(invoke_without_command=True)
 
-    parser.add_argument(
-        "--url", help="Base url of the WEBKNOSSOS instance", default=None
-    )
 
-    parser.add_argument(
-        "--jobs",
-        "-j",
-        default=DEFAULT_SIMULTANEOUS_UPLOADS,
-        type=int,
-        help=f"Number of simultaneous upload processes. Defaults to {DEFAULT_SIMULTANEOUS_UPLOADS}.",
-    )
+@app.callback()
+def main(
+    *,
+    path: Annotated[
+        Path,
+        typer.Argument(
+            show_default=False,
+            help="Path to your local WEBKNOSSOS dataset.",
+        ),
+    ],
+    url: Annotated[
+        str,
+        typer.Option(
+            help="URL to WEBKNOSSOS instance.",
+            rich_help_panel="WEBKNOSSOS context",
+            envvar="WK_URL",
+        ),
+    ] = DEFAULT_WEBKNOSSOS_URL,
+    token: Annotated[
+        Optional[str],
+        typer.Option(
+            help="Authentication token for WEBKNOSSOS instance.",
+            rich_help_panel="WEBKNOSSOS context",
+            envvar="WK_TOKEN",
+        ),
+    ] = None,
+    dataset_name: Annotated[
+        Optional[str],
+        typer.Option(
+            help="Name of your dataset on your WEBKNOSSOS.",
+            rich_help_panel="WEBKNOSSOS context",
+        ),
+    ] = None,
+    jobs: Annotated[
+        int,
+        typer.Option(
+            help="Number of simultaneous chunk uploads.",
+            rich_help_panel="Executor options",
+        ),
+    ] = DEFAULT_SIMULTANEOUS_UPLOADS,
+) -> None:
+    """Upload the WEBKNOSSOS dataset to a remote location."""
 
-    parser.add_argument(
-        "--name",
-        help="Specify a new name for the dataset. Defaults to the name specified in `datasource-properties.json`.",
-        default=None,
-    )
-
-    add_verbose_flag(parser)
-
-    return parser
-
-
-if __name__ == "__main__":
-    setup_warnings()
-    args = create_parser().parse_args()
-    setup_logging(args)
-    url = (
-        args.url
-        if args.url is not None
-        else environ.get("WK_URL", DEFAULT_WEBKNOSSOS_URL)
-    )
-    token = args.token if args.token is not None else environ.get("WK_TOKEN", None)
-    assert (
-        token is not None
-    ), f"An auth token needs to be supplied either through the --token command line arg or the WK_TOKEN environment variable. Retrieve your auth token on {url}/auth/token."
+    rprint(f"Uploading [blue]{dataset_name}[/blue] ...")
```

Comment:
The `Dataset.upload()` call already prints a progress bar, right? Therefore this information is no very useful. Please remove.